transports/quic/src/transport.rs

```diff
@@ -524,7 +540,7 @@ impl<P: Provider> Listener<P> {
                     if let Some(listen_addr) =
                         ip_to_listenaddr(&endpoint_addr, inet.addr(), self.version)
                     {
-                        log::debug!("Expired listen address: {}", listen_addr);
+                        log::debug!("Expired listen address: {listen_addr}");
```

Comment:
The `IfEvent::Down` event might be for a localhost address. In such case we would have to change `self.supports_loopback` to `false`.

I wonder whether a single wildcard listen can result in multiple localhost listen addresses. In such case setting `self.supports_loopback` to `false` would be buggy here. E.g. in case the first is down, the second might still be up.

How about tracking a `HashSet<SomeAddressType>` in `Listener`? That `HashSet` can then be iterated for loopback addresses in `eligible_listener`.

Reply:
> For example you can establish an additional network address for loopback interface by doing `ifconfig lo0 alias 127.0.0.2 up` on macOS and `if-watch` will return `127.0.0.1` and `127.0.0.2`.

Today I learned. Thank you for testing this out.